alibi/utils/wrappers.py

```diff
@@ -1,6 +1,37 @@
 import numpy as np
 
-from functools import singledispatch, update_wrapper
+from typing import Callable, Union
+
+blackbox_wrappers = {'pytorch': None, 'tensorflow': None}
+"""dict: A registry for wrappers that cast the input to a black-box function to `np.array` object and the output to a
+frame-work specific tensor.
+"""
+
+
+def blackbox_wrapper(framework: str = 'tensorflow') -> Callable:
+    """
+    A decorator that registers a wrapper for a black-box model. In the  future, the functionality may be extended so
+    that the wrapper can also be parametrized by the algorithm name, if necessary.
+
+    Parameters
+    ----------
+    framework: {'pytorch', 'tensorflow'}
+        The framework in which the the predictor wrapped is implemented.
+    """
+
+    def register_wrapper(func):
+        func.framework = framework
+        try:
+            blackbox_wrappers[framework] = func
+        except AttributeError:
+            raise AttributeError("Decorated function needs to have a framework attribute.")
+        if framework not in ['pytorch', 'tensorflow']:
+            raise ValueError(
+                f"Unknown value {framework} for decorated function attribute. Framework needs to be either 'tensorflow'"
+                f" or 'pytorch'."
+            )
+        return func
+    return register_wrapper
 
 
 class Predictor:
```

Comment:
This should be used to avoid using lambda functions as inputs (because those cannot be serialized so parallelization is not possible if you do that).